e2e/helper.bash

```diff
@@ -0,0 +1,79 @@
+setup() {
+  export TMP="$BATS_TEST_DIRNAME/../tmp"
+  export E2E_NS="e2e-$(uuidgen | tr "[:upper:]" "[:lower:]")"
+  create-namespace
+  mkdir -p "$TMP/k8s"
+  cd $BATS_TEST_DIRNAME
+}
+
+teardown() {
+  [ -f "$BATS_TEST_DIRNAME/kev.yaml" ] && rm -f "$BATS_TEST_DIRNAME/kev.yaml"
+  [ -f "$BATS_TEST_DIRNAME/docker-compose.kev.$E2E_KEV_ENV.yaml" ] && rm -rf "$BATS_TEST_DIRNAME/docker-compose.kev.$E2E_KEV_ENV.yaml"
+  [ -d "$TMP/k8s" ] && rm -rf "$TMP/k8s"
+  cd -
+}
+
+create-namespace(){
+ $KUBECTL create namespace $E2E_NS
+}
+
+generate-manifests() {
+  kev init -e $E2E_KEV_ENV && kev render -d "$TMP/k8s" -e $E2E_KEV_ENV
+}
+
+apply-manifests() {
+  $KUBECTL -n $E2E_NS apply -f "$TMP/k8s/$E2E_KEV_ENV"
+}
+
+gen-apply-manifests() {
+  generate-manifests
+  apply-manifests
+}
+
+# wait-on-deployment is responsible for waiting for a deployment to deploy
+wait-on-deployment() {
+  local namespace=$1
+  local labels=$2
+
+  for ((i=0; i<=60; i++)); do
```

Comment:
this could be re-worked slightly to make it timeout based - eg. give it 60 sec to complete the test, otherwise fail. If it fails due to timeout being not long enough then we could bump it. Currently we might wait up to 3 minutes for this to eventually fail.

Reply:
@marcinc as per your suggestion, I started on a `timeout` based solution.

But `timeout` is not cross platform - no timeout on Mac OSX … so, to keep things simple and portable I'd rather stick with this solution.

I'll try `sleep 2` to bring it down to 2 mins max.

Reply:
ok, leave as is for now. if we envisage "loop and wait" to be used in multiple places then maybe turn it into a separate function?